src/power_grid_model/core/power_grid_dataset.py

```diff
@@ -266,6 +273,18 @@ def _validate_properties(self, data: ComponentData, schema: ComponentMetaData):
         if properties.batch_size != self._batch_size:
             raise ValueError(f"Dataset must have a consistent batch size across all components. {VALIDATOR_MSG}")
 
+    def _validate_dtypes_compatibility(self, data: ComponentData, schema: ComponentMetaData):
+        sub_data = _extract_data_from_component_data(data)
+        if is_columnar(data):
+            for attr, array in sub_data.items():
+                if schema.dtype.names is None or attr not in schema.dtype.names:
+                    raise Warning(f"Attribute {attr} is not in schema. {VALIDATOR_MSG}")
+                if array.dtype != schema.dtype[attr]:
```

Comment:
For columnar data, we now have dtype of an asym attribute as dtype("float64"). 
This comes from the fact that we use `np.ascontigiousarray(data, schema_dtype[attribute].base)` for columnar data in raw data view.

Is this the right approach to keep? Will there be any other way to do it? In schema the dtype for attribute is f8 and (3,). ie. including the shape.

Reply:
i have dived into this some time ago as well. we might indeed need to work around this a little bit. an array with dtype `(f8, (3))` is different than the dtype of a column of a structured array:

```py
>>> from power_grid_model import DatasetType, ComponentType, initialize_array, power_grid_meta_data
>>> a = initialize_array(DatasetType.input, ComponentType.asym_load, 1)
>>> a.dtype.fields["p_specified"]    
(dtype(('<f8', (3,))), 16)
>>> a["p_specified"].dtype
dtype('float64')
>>> power_grid_meta_data[DatasetType.input][ComponentType.asym_load].dtype["p_specified"].base
dtype('float64')
```

the best we can do is check that the last dimension is equal to `3`

even worse (in our case): 
```py
>>> np.array(a["p_specified"], dtype=("<f8", (3,))).shape
(1, 3, 3)
>>> a["p_specified"].shape 
(1, 3)
```